### PR TITLE
Bump tap version

### LINKED ISSUE #802: Update agent dependencies for node 16 support
Update package.json to include new major version updates for the following packages:

 * `@newrelic/aws-sdk`
 * `@newrelic/koa`
 * `@newrelic/native-metrics`
 * `@newrelic/superagent`
 * `@newrelic/test-utilities`

### PR DESCRIPTION
**Pull request by bizob2828**
## Proposed Release Notes
 * Upgraded tap to v15
 * Upgraded `@newrelic/superagent` `@newrelic/aws-sdk` `@newrelic/koa` `@newrelic/native-metrics` and `@newrelic/test-utilities` to the latest

## Links
Closes #802

## Details
This needs to land before we can re-test https://github.com/newrelic/node-newrelic/pull/772

### CODE REVIEW

File: test/unit/instrumentation/http/http.test.js [hunk 1]
```diff
@@ -292,6 +283,8 @@ test('built-in http module instrumentation', (t) => {
           const segment = transaction.baseSegment
           const spanAttributes = segment.attributes.get(DESTINATIONS.SPAN_EVENT)
 
+          // TODO: these attrs aren't getting filtered
+          // not familiar with code so trying to trace where this is supposed to be removed
```
Comment (carlo-808) [inline test/unit/instrumentation/http/http.test.js:287]: I believe the `x-filtered-out` header is getting filtered out through attribute filtering wildcard rules. Looks like the regex test is `/^(?:response\.(?:headers\.(?:setCookie|x))|request\.(?:headers\.(?:setCookie|x)))/g` for filtering out.

**Review by carlo-808 — COMMENTED** [review]
Looks good so far. Just a few really minor comments. Waiting for rebase to approve.

File: test/unit/instrumentation/http/http.test.js (hunk 1 above)
Comment (bizob2828) [reply test/unit/instrumentation/http/http.test.js:287]: yea i forgot to remove TODO after i traced that

**Review by carlo-808 — APPROVED** [review]
Tap updated to 15 🎉 with a lot of test updates (nice work)
Node module deps updated 🎉

**Pull request merged**